### PR TITLE
feat(npm-event): allow for larger npm packages

### LINKED ISSUE #504: Greenkeeper Hook Endpoint Responds with HTTP 413 when Updating a npm Package
As described in the initial pull request of greenkeeper 2 I tried to register a hook for private npm repositories with 

`wombat hook add @<package>/<name> https://hooks.greenkeeper.io/npm/<id> <token>`

After publishing a new version at npm I checked the HTTP Response with wombat. `wombat hooks ls` responded with a HTTP 413 status, which means that the request size was too big.

In our private repository we have a lot of stuff and the size of the package.json/shrinkwrap is around 300kb (I assume this will be sent to greenkeeper). I guess this is too much.

Could you please make the accepted request size of the hook endpoint bigger (`https://hooks.greenkeeper.io/npm/<id>`)?

### PR DESCRIPTION
**Pull request by Realtin**
resolves https://github.com/greenkeeperio/greenkeeper/issues/504

### CODE REVIEW

File: lib/npm-event.js [hunk 1]
```diff
@@ -17,7 +17,8 @@ function npmEvent (server, {env, channel}, next) {
     config: {
       payload: {
         output: 'data',
-        parse: false
+        parse: false,
+        maxBytes: 26214400
```
Comment (janl) [inline lib/npm-event.js]: It is good practice to document what numbers mean. `maxBytes` is a good indicator of what we are looking at, but a small comment like `maxBytes: 26214400 // 24 * 1024 * 1024 = 24MB` can go a long way in longer-term maintainability. (hope this doesn’t sound too condescending ;D)

File: test/npm-event.js [hunk 2]
```diff
@@ -94,4 +94,45 @@ const register = require('../lib/npm-event')
...
+    .update(reqPayload)
+    .digest('hex')
+
+    const {statusCode} = await server.inject({
+      method: 'POST',
+      url: `/npm/${installation}`,
+      headers: {
+        'x-npm-signature': `sha256=${hmacPayload}`,
+        'Content-Type': 'application/json',
+        'Content-Length': 20000000
```
Comment (janl) [inline test/npm-event.js]: Correct me if I’m mistaken, but this is only testing that when we *say* that the payload is `20000000` bytes, but the request body is the JSON object in lines 99-107? — Shouldn’t we test with a real 1MB+ body?

Also, see comment above about large numbers :)

**Review by janl — CHANGES_REQUESTED** [review]
looks good overall, just two small notes

File: test/npm-event.js (hunk 2 above)
Comment (Realtin) [reply test/npm-event.js]: How do I get a 1MB+ body?

Comment (janl) [reply test/npm-event.js]: Something like this maybe? `new Buffer.alloc(1024 * 1024 * 2).toString()` for 2MB, and then stick that in an unused field on the body object?

**Review by janl — APPROVED** [review]

**Review by gr2m — APPROVED** [review]
changes look good to me, but CI fails. Let me know if I can be of help debugging :)

File: test/npm-event.js [hunk 3]
```diff
@@ -94,4 +94,46 @@ const register = require('../lib/npm-event')
     t.same(job.properties.priority, 1, 'job priority')
     t.end()
   })
+
+  tap.test('accepts packages that are bigger then 1MB', async (t) => {
+    const bigBody = Buffer.alloc(1024 * 1024 * 5).toString()
```
Comment (Realtin) [inline test/npm-event.js]: it is fine up until `1024 * 1024 * 4.1`. Starting to fail with `1024 * 1024 * 4.2`. Is there some magical number inbetween?

Comment (Realtin) [reply test/npm-event.js]: "Payload content length greater than maximum allowed: `26 214 400`"

but  1024 * 1024 * 4.2 = `4 404 019.2` ... but (!) after JSON.stringify-ing  the reqPayload is `26 216 594`  the size of the body multiplied by more than 6!

**Pull request merged**